libraries/AP_BLHeli/AP_BLHeli.h

```diff
@@ -40,9 +42,29 @@ class AP_BLHeli {
     bool process_input(uint8_t b);
 
     static const struct AP_Param::GroupInfo var_info[];
+
+    struct telem_data {
+        uint8_t temperature; // degrees C
+        uint16_t voltage;    // volts * 100
+        uint16_t current;    // amps * 100
+        uint16_t consumption;// mAh
+        uint16_t rpm;        // eRPM
+        uint16_t count;
+    };
+
+    // get the most recent telemetry data packet for a motor
+    bool get_telem_data(uint8_t esc_index, struct telem_data &td, uint32_t &timestamp_ms);
+
+    static AP_BLHeli *get_instance(void) {
```

Comment:
These should be called singleton to be consistent with the rest of the code base. (And it's less confusing, especially for devices that may have multiple instances)

Reply:
well, I can change it if you like, but we have more get_instance() than get_singleton() in the code right now

Reply:
changed